src/line2d.jl

```diff
@@ -104,3 +104,50 @@ function xiaolin_wu{T<:Gray}(img::AbstractArray{T, 2}, y0::Int, x0::Int, y1::Int
     end
     img
 end
+
+"""
+```
+img_with_line = line_normal(img, τ, color, method)
+img_with_line = line_normal(img, ρ, θ, color, method)
+```
+
+Draws a line on the input image given tuple `τ` of the form (ρ,θ) for the normal form of line:
+    x*cos(θ) + y*sin(θ) = ρ
+with the given `color`. Lines are drawn using the `bresenham` method by default. If anti-aliasing
+is required, the `xiaolin_wu` can be used.
+"""
+line_normal{T<:Colorant}(img::AbstractArray{T, 2}, args...) = line_normal!(copy(img), args...)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, τ::Tuple{Number,Number}, method::Function = bresenham) = 
+    line_normal!(img, τ[1], τ[2], one(T), method)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, τ::Tuple{Number,Number}, color::T, method::Function = bresenham) = 
+    line_normal!(img, τ[1], τ[2], color, method)
+
+line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, ρ::Number, θ::Number, method::Function = bresenham) =
+    line_normal!(img, ρ, θ, one(T), method)
+
+function line_normal!{T<:Colorant}(img::AbstractArray{T, 2}, ρ::Number, θ::Number, color::T, method::Function = bresenham)
+    h,w = size(img)
+    ρ -= 1; h -= 1; w -= 1
+    t = Vector{Number}(4)
+    cosθ = cos(θ)
+    sinθ = sin(θ)
+    t[1] = ( ρ * cosθ    )/ sinθ
```

Comment:
I wonder if it would be clearer to compute the 4 intersection points between the (extended) line and the (extended) lines that form the boundaries of the image:
```julia
intersections_x = [(x, (ρ - x*cosθ)/sinθ) for x in (first(indsx), last(indsx))] # with the left and right edge lines
intersections_y = [((ρ - y*sinθ)/cosθ, y) for y in (first(indsy), last(indsy))] # with the top and bottom edge lines
```
and then keeping the 0 or 2 intersections that occur within the bounds of the image. (What happens with the current implementation if `ρ, θ` are such that the line doesn't intersect the image?)

If you like this parametrization, note there are no subtractions of 1 like on line 132.